spec/core/ics-003-connection-semantics/UPGRADES.md

```diff
@@ -0,0 +1,483 @@
+# Upgrading Connections
+
+### Synopsis
+
+This standard document specifies the interfaces and state machine logic that IBC implementations must implement in order to enable existing connections to upgrade after the initial connection handshake.
+
+### Motivation
+
+As new features get added to IBC, chains may wish the take advantage of new connection features without abandoning the accumulated state and network effect(s) of an already existing connection. The upgrade protocol proposed would allow chains to renegotiate an existing connection to take advantage of new features without having to create a new connection, thus preserving all existing channels that built on top of the connection.
+
+### Desired Properties
+
+- Both chains MUST agree to the renegotiated connection parameters.
+- Connection state and logic on both chains SHOULD either be using the old parameters or the new parameters, but MUST NOT be in an in-between state, e.g., it MUST NOT be possible for a chain to write state to an old proof path, while the counterparty expects a new proof path.
+- The connection upgrade protocol is atomic, i.e., 
+  - either it is unsuccessful and then the connection MUST fall-back to the original connection parameters; 
+  - or it is successful and then both connection ends MUST adopt the new connection parameters and process IBC data appropriately.
+- The connection upgrade protocol should have the ability to change all connection-related parameters; however the connection upgrade protocol MUST NOT be able to change the underlying `ClientState`.
+The connection upgrade protocol MUST NOT modify the connection identifiers.
+
+## Technical Specification
+
+### Data Structures
+
+The `ConnectionState` and `ConnectionEnd` are defined in [ICS-3](./README.md), they are reproduced here for the reader's convenience. `UPGRADE_INIT`, `UPGRADE_TRY`, `UPGRADE_ERR` are additional states added to enable the upgrade feature.
```

Comment:
Is `UPGRADE_ERR` still needed? It is missing below

Reply:
Thanks it was removed